stubs/pyjks/jks/util.pyi

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Iterable, Mapping
+from pathlib import Path
+from struct import Struct
+from typing import Any
+from typing_extensions import Final, Literal, Self, TypeAlias
+
+from .bks import BksKeyEntry
+from .jks import PrivateKeyEntry
+
+b8: Final[Struct]
+b4: Final[Struct]
+b2: Final[Struct]
+b1: Final[Struct]
+py23basestring: Final[tuple[type[str], type[str]]]
+RSA_ENCRYPTION_OID: Final[tuple[int, ...]]
+DSA_OID: Final[tuple[int, ...]]
+DSA_WITH_SHA1_OID: Final[tuple[int, ...]]
+
+_KeystoreType: TypeAlias = Literal["jks", "jceks", "bks", "uber"]
+_PemType: TypeAlias = Literal["CERTIFICATE", "PUBLIC KEY", "PRIVATE KEY", "RSA PRIVATE KEY"]
+
+class KeystoreException(Exception): ...
+class KeystoreSignatureException(KeystoreException): ...
+class DuplicateAliasException(KeystoreException): ...
+class NotYetDecryptedException(KeystoreException): ...
+class BadKeystoreFormatException(KeystoreException): ...
+class BadDataLengthException(KeystoreException): ...
+class BadPaddingException(KeystoreException): ...
+class BadHashCheckException(KeystoreException): ...
+class DecryptionFailureException(KeystoreException): ...
+class UnsupportedKeystoreVersionException(KeystoreException): ...
+class UnexpectedJavaTypeException(KeystoreException): ...
+class UnexpectedAlgorithmException(KeystoreException): ...
+class UnexpectedKeyEncodingException(KeystoreException): ...
+class UnsupportedKeystoreTypeException(KeystoreException): ...
+class UnsupportedKeystoreEntryTypeException(KeystoreException): ...
+class UnsupportedKeyFormatException(KeystoreException): ...
+
+class AbstractKeystore:
+    store_type: _KeystoreType
+    entries: Mapping[str, AbstractKeystoreEntry]
+    def __init__(self, store_type: _KeystoreType, entries: dict[str, AbstractKeystoreEntry]) -> None: ...
+    @classmethod
+    def load(cls, filename: str | bytes | Path, store_password: str | None, try_decrypt_keys: bool = True) -> Self: ...
+    def save(self, filename: str | bytes | Path, store_password: str) -> None: ...
+
+class AbstractKeystoreEntry:
+    store_type: _KeystoreType | None
+    alias: str
+    timestamp: int
+    def __init__(self, *, store_type: _KeystoreType = ..., alias: str = ..., timestamp: int = ..., **kwargs: Any) -> None: ...
+    @classmethod
+    def new(cls, alias: str) -> Self: ...
+    def is_decrypted(self) -> bool: ...
+    def decrypt(self, key_password: str) -> None: ...
+    def encrypt(self, key_password: str) -> None: ...
+
+def as_hex(ba: bytes | bytearray) -> str: ...
+def as_pem(der_bytes: bytes, type: _PemType) -> str: ...
+def bitstring_to_bytes(bitstr: Iterable[Literal[0, 1]]) -> bytes: ...
```

Comment:
This annotation might be pretty annoying in practice. I'm not sure it's ever going to be possible to narrow the type of something to `Iterable[Literal[0, 1]]` without using a `cast()`. Could consider using `Iterable[int]` instead:

```suggestion
def bitstring_to_bytes(bitstr: Iterable[int]) -> bytes: ...
```